src/main/java/com/gluonhq/connect/converter/JsonConverter.java

```diff
@@ -113,13 +111,17 @@ public T readFromJson(JsonObject json) {
         T t = null;
 
         try {
-            t = targetClass.newInstance();
+            if (! "java.util.Map".equals(targetClass.getName())) {
```

Comment:
Check the class directly with: `Map.class.isAssignableFrom(targetClass)`? That also covers the case when any of the subclasses of Map are being used.

Reply:
done